crates/blockchain-tree/src/shareable.rs

```diff
@@ -109,6 +109,14 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTreeViewer
         self.tree.read().block_by_hash(block_hash).cloned()
     }
 
+    fn buffered_block_by_hash(&self, block_hash: BlockHash) -> Option<SealedBlock> {
+        self.tree.read().get_buffered_block(&block_hash).map(|b| b.block.clone())
+    }
+
+    fn buffered_header_by_hash(&self, block_hash: BlockHash) -> Option<SealedHeader> {
+        self.tree.read().get_buffered_block(&block_hash).map(|b| b.header.clone())
+    }
+
```

Comment:
Nit I'd make this return reference instead of cloning

Reply:
not possible unfortunately because can't borrow past read lock

Reply:
ah yikes - good point